### PR TITLE
Integrating ZeroMQ and Simple Python Demo

### PR DESCRIPTION
**Pull request by rayaq-siddiqui**
Created PR for the ZeroMQ work Jordan started. This is going to be the end to end work for the simple sorting algorithm with the python sorting and c++ peer to peer communication.

### CODE REVIEW

**Comment by rayaq-siddiqui** [comment]
It seems like the "empty" files are added because of permission changes in the repo. So it is not relevant to the PR and not causing changes. The non-empty files are the only impactful changes in this repo.

File: src/Peers/provider.cpp [hunk 1]
```diff
@@ -53,6 +54,8 @@ void Provider::listen() {
             continue;
         }
 
+        // zmq_sender.send("P2P: task request received!");
```
Comment (PalaashKolhe) [inline src/Peers/provider.cpp]: nit: comment left behind

File: src/Peers/provider.cpp [hunk 2]
```diff
@@ -175,19 +178,72 @@ void Provider::followerHandleTaskRequest() {
 void Provider::processWorkload() {
     // data is stored in task->training data
     vector<int> data = task->getTrainingData();
-    sort(data.begin(), data.end());
+    // sort(data.begin(), data.end());
```
Comment (PalaashKolhe) [inline src/Peers/provider.cpp]: nit: comment

File: third_party/MODULE.bazel
Comment (PalaashKolhe) [inline third_party/MODULE.bazel:1]: is this file used? if not, can we get rid of it?

**Review by PalaashKolhe — CHANGES_REQUESTED** [review]

File: third_party/MODULE.bazel
Comment (rayaq-siddiqui) [reply third_party/MODULE.bazel:1]: Yes, required to be able to recognize it as a Bazel module. I tried removing it and it would not build.

**Review by TonyxSun — APPROVED** [review]
I am able to build and run this with 3 providers; great work!!!!!

**Review by PalaashKolhe — APPROVED** [review]

**Pull request merged**